src/infra/src/table/folders.rs

```diff
@@ -0,0 +1,107 @@
+use config::meta::folder::Folder;
+use sea_orm::{
+    ActiveModelTrait, ActiveValue::NotSet, ColumnTrait, DatabaseConnection, EntityTrait,
+    ModelTrait, QueryFilter, QueryOrder, Set,
+};
+
+use super::entity::folder::{ActiveModel, Column, Entity, Model};
+use crate::{
+    db::{connect_to_orm, ORM_CLIENT},
+    errors,
+};
+
+// Flags that indicate the type of data the folder can contain.
+const DASHBOARDS_FOLDER_TYPE: i16 = 0;
+
+impl From<Model> for Folder {
+    fn from(value: Model) -> Self {
+        Self {
+            folder_id: value.folder_id,
+            name: value.name,
+            description: value.description.unwrap_or_default(),
+        }
+    }
+}
+
+/// Gets a folder by its `folder_id` surrogate key.
+pub async fn get(org_id: &str, folder_id: &str) -> Result<Option<Folder>, errors::Error> {
+    let client = ORM_CLIENT.get_or_init(connect_to_orm).await;
```

Comment:
💡 Something for future consideration:

In many of our "DB access" functions in the `infra` crate we initialize a lazy static DB connection and then use that to perform DB operations.

Although this functions, it is limited in that it ties the implementation of the function to a single concrete type of the DB connection and, therefore, it makes unit testing these functions infeasible since we can't easily swap the real DB connection out for a mock DB connection that can be used and interrogated in unit tests.

An alternative approach would be change the type signature of "DB access" functions (and their callers) to accept a `client` parameter whose type is either a generic type that implements a "DB connection trait" or a trait object for that trait. The actix router would, upon initialization, create this connection and clone and pass it to its route-handling threads. Then to unit test these "DB access" functions we could create a `MockDbConnection` type that implements the "DB connection trait" and then use that mock connection in the unit tests.

Certainly not urgent or necessary, but if we get to the point in the future where we want to unit test these functions it might be worth considering this.

Reply:
great suggestion. we can work on that as a separate project.